### PR TITLE
Added `QUEUE_ID` to settings

### LINKED ISSUE #350: `mef_eline` should allow to set on a default queue to be used when an EVC is created
This is for introducing a default `QUEUE_ID` that should be used when an EVC is first created. It should be backwards compatible, so on settings.py the default value should be `QUEUE_ID = None`, but network operators might start to use a different value.

### PR DESCRIPTION
**Pull request by Alopalao**
Closes #350 

### Summary

- Added `QUEUE_ID` to `settings.py` to be a default value for EVCs `"queue_id"` when created
- Changed UI to for `queue_id`, it is now a `k-dropdown` showing values `"default"`, `"none"`, and range(0,8).
- `openapi.yml` is allowing `-1` as a value for `queue_id`.
- When creating an EVC, if `queue_id` is not present or has `-1`, it is taken to be default now.
- Changing the value from `settings.py` is not going to trigger a redeployment. If a redeployment happens, `queue_id` will be updated with the current default value.

### Local Tests

Change `QUEUE_ID` value, through API requests and UI
Create an EVC, through API requests and UI

### CODE REVIEW

File: settings.py
Comment (viniarck) [inline settings.py:53]: Could you add a coment above this line summarizing what's it's for, and the supported values? We do have that docs on the openapi.yml but, it's also convenient to have here just so it's immediately clear for whoever is setting the settings.py file (same pattern we've been following for the other exposed vars)

**Review by viniarck — COMMENTED** [review]
@Alopalao overall looks good, that's what we needed.

Could you also adapt the front-end a find a solution accordingly? and explore it both on `k-toolbar` and `k-info-panel`? In the front-end, we also need to capture a semantics of default value (maybe -1, an out of range value? and then display it as `"default"` in the front-end). The tricky part in this case is that a `queue_id`: `null/None` value currently already carries the semantics of "no queue" (that you've fixed recently), so we'll need to rely on another value to represent the default one, and whenever this default value is received then when an EVC is being created or edited, then it should be mapped accordingly. See what I mean? For `spf_attribute` I had to solve a similar problem [on PR 355](https://github.com/kytos-ng/mef_eline/pull/355), but there the semantics of `null/None` was easier, since `spf_attribute` `None` didn't have a semantics of not using spf_attribute unlike `queue_id`.

**Comment by viniarck** [comment]
> @Alopalao overall looks good, that's what we needed.
> 
> Could you also adapt the front-end a find a solution accordingly? and explore it both on `k-toolbar` and `k-info-panel`? In the front-end, we also need to capture a semantics of default value (maybe -1, an out of range value? and then display it as `"default"` in the front-end). The tricky part in this case is that a `queue_id`: `null/None` value currently already carries the semantics of "no queue" (that you've fixed recently), so we'll need to rely on another value to represent the default one, and whenever this default value is received then when an EVC is being created or edited, then it should be mapped accordingly. See what I mean? For `spf_attribute` I had to solve a similar problem [on PR 355](https://github.com/kytos-ng/mef_eline/pull/355), but there the semantics of `null/None` was easier, since `spf_attribute` `None` didn't have a semantics of not using spf_attribute unlike `queue_id`.

> value (maybe -1, an out of range value? and then display it as "default")

@Alopalao, actually, thinking more about it, a cleaner approach could potentially be in the data model:

`queue_id` undefined (when this key isn't set) means default
`queue_id` None means no queue
`queue_id` integer (0-7) means an actual queue number

That way we if we differentiate between undefined (when this key isn't set), and when it's actually None then it would behave the same way as we have for `spf_attribute` in the runtime in memory. I have an impression that it'd be easier for the front-end too, simplifying for the caller, since essentially you wouldn't need to map between values in the front-end (and also we wouldn't need to extend the range of the API values), and we'd only have to be careful when to send `queue_id` or not, as long as in the back-end projection for `queue_id` we also don't return this key when it's not set.

Could you double check this idea, or see if you can come up with a simpler one?

File: settings.py
Comment (Alopalao) [reply settings.py:55]: Comment added

**Comment by Alopalao** [comment]
- Changed UI to for `queue_id`, it is now a `k-dropdown` showing values `"default"`, `"none"`, and range(0,8).
- `openapi.yml` is allowing `-1` as a value for `queue_id`.
- When creating an EVC, if `queue_id` is not present or has `-1`, it is taken to be default now.
- Changing the value from `settings.py` is not going to trigger a redeployment. If a redeployment happens, `queue_id` will be updated with the current default value.

File: ui/k-toolbar/main.kytos
Comment (viniarck) [inline ui/k-toolbar/main.kytos:289]: Can we have the default selected here? Just so in the front-end by default it also follows the default? We've been using this way for [`spf_attribute`](https://github.com/kytos-ng/mef_eline/blob/master/ui/k-toolbar/main.kytos#L301)

File: ui/k-info-panel/show_circuit.kytos
Comment (viniarck) [inline ui/k-info-panel/show_circuit.kytos:68]: Great enhancement. Awesome to have this `k-dropdown`.

File: openapi.yml
Comment (viniarck) [inline openapi.yml:446]: Since `queue_id` isn't mandatory, essentially the caller would have two ways to set a default, either not sending `queue_id` or setting `queue_id: -1`, which opens a bit of ambiguity, which is reasonable since it also doesn't break backwards compatibility, but could we get away with only allowing to set a default when `queue_id` isn't set and then not changing the API here with `-1`, and then only using `-1` in the runtime `EVC` but never actually storing it in the DB, and [adapt the `queue_id` projection](https://github.com/kytos-ng/mef_eline/blob/master/db/models.py#L164) allowing it to be undefined (not included)? Just so it follows a same pattern that we have for `spf_attribute`, and then we can keep this pattern if any other attribute also needs a dynamic default? 

I wouldn't reject this PR as it is, I think it's reasonable, but following the existing pattern would be great, I'll leave it up to you and @italovalcy to reach a consensus what you both prefer since you've been already aligning/discussing the requirements, great stuff, @Alopalao.

Comment (Alopalao) [reply openapi.yml:446]: Talked to Italo and decided to go with `-1`. But for completeness, there is a limitation if we do not want an extra value. The updates are different from UI and API:
- The UI sends a completed dictionary of an EVC. If a field is not present in it, we can assure that this non-existent value should be removed. This way we make a difference between non-existent, `None`, and integer.
- The API request presents the possibility to have partial updates. We cannot assure the desire to delete a field if it is not in the dictionary from the request. In this case, we need an extra value to delete `queue_id`.
We can differentiate two types of updates (complete, partial) but since this PR is not that serious I think it is not necessary.
If we have another case where non-existent and `None` should be different, this complication will appear again.

File: ui/k-toolbar/main.kytos
Comment (Alopalao) [reply ui/k-toolbar/main.kytos]: Changed.

File: ui/k-info-panel/show_circuit.kytos
Comment (Alopalao) [reply ui/k-info-panel/show_circuit.kytos:68]: Thanks

File: openapi.yml
Comment (viniarck) [reply openapi.yml:446]: Exactly, Aldo. 

Let's go with `-1` then since you guys are on the same page.

Before we close this discussion, can you also document the `-1` in the openapi.yml description, and set it as the default value in the spec too?

Comment (Alopalao) [reply openapi.yml]: Added comments.

Comment (Alopalao) [reply openapi.yml]: Added comments.

File: openapi.yml
Comment (viniarck) [inline openapi.yml:446]: @Alopalao can you move the comment to the description, what's what the description is for, just so it can also be rendered nicely.

**Review by viniarck — APPROVED** [review]
I'll leave the PR pre-approved @Alopalao. 

But let's only merge this once `2023.1` is relased since we're on code freeze. Also, it'd be great to have @italovalcy review before merging since you both were discussing and collaborating on this feature. 

Nicely done.

File: openapi.yml
Comment (Alopalao) [reply openapi.yml]: Description moved.

**Pull request merged**
